File: A's/cpp/141A.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    string guest, host, pile;
    cin >> guest >> host >> pile;

    string combined = guest + host;
    
    if (combined.size() != pile.size()) {
        cout << "NO" << endl;
        return 0;
    }

    sort(combined.begin(), combined.end());
    sort(pile.begin(), pile.end());

    if (combined == pile) {
        cout << "YES" << endl;
    } else {
        cout << "NO" << endl;
    }
    
    return 0;
}

File: A's/py/141A.py
def main() -> None:
    guest: str = str(input())
    host: str = str(input())
    pile: str = str(input())
    combined: str = guest + host

    if len(combined) != len(pile):
        print("NO")
    else:
        combined = sorted(combined)
        pile = sorted(pile)

        if combined == pile:
            print("YES")
        else:
            print("NO")

if __name__ == "__main__":
    main()